### DISCUSSION TITLE
Make an alternative to `miryoku_kmonad`'s `MIRYOKU_MAPPING=LITE` that is non-hybrid and has better corresponding thumb keys

### DISCUSSION THREAD

**Discussion by hab25** [manna-harbour/miryoku#163] [Ideas]
As the current [best](https://www.reddit.com/r/ErgoMechKeyboards/comments/v091fk/comment/iahijvl/?utm_source=share&utm_medium=web2x&context=3) option for general Miryoku use on common row-staggered keyboards, [MIRYOKU_MAPPING=LITE](https://github.com/manna-harbour/miryoku_kmonad#lite) could be improved with regards to:
- the corresponding thumb keys

	problems:
	- except for `space` and `tab`, the corresponding thumb keys require too much pinky finger movement
	- `del`, `esc` often (e.g. space-constrained keyboards) have variation in their location

 	improvements: 
  	- `enter` goes to `open square bracket`
  	- `del` goes to `right shift`
  	- `backspace` goes to `quote`
  	- `escape` goes to `caps lock`
	
	These were decided by choosing the least-pinky-movement keys. The specific pairings were made by following the same ordering of vanilla myrioku's thumb keys, but going up-to-down instead of left-to-right. Note that backspace, which on vanilla miryoku is the home for right thumb, is here the easiest to type out of the corresponding right thumb keys. 
	Space was left as is: the thumb types better than the pinky.
	Tab was left as is: since it lines up with the common tab position, it makes the transition easier.

- the hybridness 
	
	problems:
	-  it is possible to press the non-miryoku keys by accident,
	-  it reduces the area where one can rest their hand on their common keyboard
	
	improvement:
	-  disable all keys which are already present in some myrioku layer (i.e. all of the ones which are not part of the 36)

**Comment by hab25** [comment]
@manna-harbour do you think this is a good idea? I failed to clarify that this is all up for discussion.

I think this would a big improvement for miryoku usage on laptop keyboards.

**Comment by manna-harbour** [comment]
Thanks for the suggestion!

So, the corresponding thumb keys are really only meant as an optional additional feature, to give some kind of access to keys that aren't included in the native keyboard layout. But the intention for normal use is that all the keys you need are in the native layout and that the mapping only adds alternative alpha options and convenient access to Nav and mods. 

Other hybrid mappings could be added that give more convenient access to the rest of the thumb keys at the expense of losing the native function of that key position, but they will need to be tested to see if the compromises are worthwhile. Using the next column on each side as you've done seems like a good choice for this. But of course using the pinkie for layers means you lose normal access to Super (as happens now with layer-tap access to the Button layer), and combining layer change with mods might be an uncomfortable stretch. I'd probably favour `Tab` / `Caps` / `Left Shift` and `]` / `'` / `Right Shift` for symmetry with the flip layers option (plus `Space` of course). This could be a hybrid mapping or a regular mapping with no extra native layout keys.

Miryoku KMonad doesn't have [customisation support](https://github.com/manna-harbour/miryoku/discussions/85) for mappings, but you could just edit [the lite mapping thumb keys](https://github.com/manna-harbour/miryoku_kmonad/blob/21fd0cf290b2556a3579a67f851a34230b2053bb/src/miryoku_kmonad.kbd.cpp#L37) directly if you'd like to try out your ideas.

Another option for a normal mapping (for [60 ansi](https://github.com/qmk/qmk_firmware/blob/master/layouts/default/60_ansi/default_60_ansi/keymap.c)) is to home on `D` and `L` (shift left hand left one column and right hand right 2 columns) and use `Gui` / `L Alt` / `Space` and `R Alt` / `R Gui` / `Menu` for thumbs. Also I'm planning on adding another similar mapping for use with row stagger boards that do have enough well placed thumb keys to use Miryoku normally, such as Japanese laptop keyboards.

**Comment by hab25** [comment]
> But of course using the pinkie for layers means you lose normal access to Super

Ah, great point, can't believe I didn't see this. For me, this is too large of a drawback and thus I consider my suggestions to be invalid.

> I'd probably favour Tab / Caps / Left Shift and ] / ' / Right Shift for symmetry with the flip layers option (plus Space of course).

Reasonable.

> Another option for a normal mapping (for [60 ansi](https://github.com/qmk/qmk_firmware/blob/master/layouts/default/60_ansi/default_60_ansi/keymap.c)) is to home on D and L (shift left hand left one column and right hand right 2 columns) and use Gui / L Alt / Space and R Alt / R Gui / Menu for thumbs.

This looks good! It seems that it would work for me on all the keyboards I care about. What are your concerns with this one, given that you haven't added it to Miryoku? Perhaps you consider those thumb keys to be badly placed and too awkward to use?

**Reply by manna-harbour** [reply]
> Ah, great point, can't believe I didn't see this. For me, this is too large of a drawback and thus I consider my suggestions to be invalid.

Depending on how you use Super it might be ok. It's similar to what's done already for [reviung39](https://github.com/manna-harbour/miryoku_zmk/blob/master/miryoku/mapping/39/reviung39.h) due to lack of thumb keys. But it's not a complete solution. If you need Super on a layer you could use the other hand (if not using the pinkie there) or enable sticky keys on the OS.

> Perhaps you consider those thumb keys to be badly placed and too awkward to use?

That was probably the original reason, but looking at it now it seems ok. If the tertiary thumb is too far it could be used with [thumb combos](https://github.com/manna-harbour/miryoku/issues/56). Combos aren't supported in KMonad but [𝑥MK](https://github.com/manna-harbour/xmk) could be used instead.

**Reply by hab25** [reply]
> Depending on how you use Super it might be ok.

True, but I value flexibility and don't want to constrain myself to not being able to use super in certain ways even if I don't currently have plans for doing so.
> enable sticky keys on the OS

Maybe this could work, but my first impression here is that there would be too much thinking and too many keypresses; I don't think would ever be seamless for me.
>That was probably the original reason, but looking at it now it seems ok.

Great! I'll be trying it out soon. Do you plan to implement it in Miryoku KMonad?

**Reply by manna-harbour** [reply]
> Great! I'll be trying it out soon. Do you plan to implement it in Miryoku KMonad?

Let me know if you try it. I can add it to Miryoku KMonad and 𝑥MK if it's usable.

The row stagger board I have here has 4x1u keys to the left of space and it seems like this mapping would be ok, but looking at the standard 60 ansi layout with 3x1.25u it looks to be too much of a stretch for the thumb.
